pkg/services/pluginsintegration/pluginsintegration.go

```diff
@@ -146,6 +149,14 @@ func NewClientDecorator(
 	c := client.ProvideService(pluginRegistry, pCfg)
 	middlewares := CreateMiddlewares(cfg, oAuthTokenService, tracer, cachingService, features)
 
+	log.RegisterContextualLogProvider(func(ctx context.Context) ([]any, bool) {
```

Comment:
Since `WithContextualAttributes` is in the log package I would recommend putting this in there as well/together with it in the same package and by so you don't need to expose `log.LogParamsContextKey`.